### PR TITLE
#71 - Add auth IT for repository HTTP

### LINKED ISSUE #71: Authentication and authorization
PHP composer does not have auth support yet and all operations are available to all users. It should have authentication support and require `read` and `write` permissions for it's endpoints

### PR DESCRIPTION
**Pull request by genryxy**
Closes #71 
Added auth IT for repository HTTP

### CODE REVIEW

File: src/test/java/com/artipie/composer/http/RepositoryHttpAuthIT.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        Logger.debug(this, "Command:\n%s\n", String.join(" ", command));
+        final Container.ExecResult res = this.cntn.execInContainer(command);
+        final String log = String.format(
+            "STDOUT:\n%s\nSTDERR:\n%s", res.getStdout(), res.getStderr()
+        );
+        Logger.debug(this, log);
+        return log;
+    }
+
+    private void start(final TestAuthentication.User user) {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/composer/http/RepositoryHttpAuthIT.java:210]: @genryxy I'd suggest the following change: move everything except url creation from this method into `setUp()` and rename this method into `url()` and make it return `String` url. Field `url` can be removed then

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy please, check my comment. Also, I do not completely follow how is it possible, that we are using two different storages here? We add package to one in `addPackage()` and `upload()` methods and install from the other one created in `start()` method...

**Comment by genryxy** [comment]
@olenagerasimova `upload()` is necessary to create remote server from which the package will be uploaded. And url for package is specified in `dist` section in `PackageSimple` class. The second storage is used for `PhpComposer` slice

**Comment by olenagerasimova** [comment]
@genryxy thanks, got it, could you please still consider suggestion from review?

**Comment by genryxy** [comment]
@olenagerasimova thanks, made changes

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks

**Comment by 0crat** [comment]
Job `gh:artipie/composer-adapter#82` is not assigned, can't get performer

**Pull request merged**